shared/chat/conversation/messages/shared.desktop.js

```diff
@@ -0,0 +1,134 @@
+// @flow
+import React, {PureComponent} from 'react'
+import {Avatar, Icon, Text} from '../../../common-adapters'
+import {globalStyles, globalMargins, globalColors} from '../../../styles'
+import * as Constants from '../../../constants/chat'
+import {withHandlers} from 'recompose'
+import shallowEqual from 'shallowequal'
+
+type Props = {
+  includeHeader: boolean,
+  isFirstNewMessage: boolean,
+  onRetry: () => void,
+  onAction: (message: Constants.ServerMessage, event: any) => void,
+  style: Object,
+  isSelected: boolean,
+  children: React$Element<*>,
+  message: Constants.TextMessage | Constants.AttachmentMessage,
+}
+
+const _marginColor = (followState) => ({
+  'You': globalColors.white,
+  'Following': globalColors.green2,
+  'NotFollowing': globalColors.blue,
+  'Broken': globalColors.red,
+}[followState])
+
+const colorForAuthor = (followState: Constants.FollowState) => {
+  if (followState === 'You') {
+    return globalColors.black_75
+  } else {
+    return _marginColor(followState)
+  }
+}
+
+const Retry = ({onRetry}: {onRetry: () => void}) => (
+  <div>
+    <Text type='BodySmall' style={{fontSize: 9, color: globalColors.red}}>{'┏(>_<)┓'}</Text>
+    <Text type='BodySmall' style={{color: globalColors.red}}> Failed to send. </Text>
+    <Text type='BodySmall' style={{color: globalColors.red, textDecoration: 'underline'}} onClick={onRetry}>Retry</Text>
+  </div>
+)
+
+type MessageProps = Props & {onIconClick: (event: any) => void}
+
+class _MessageComponent extends PureComponent<void, MessageProps, void> {
+  shouldComponentUpdate (nextProps: MessageProps) {
+    return !shallowEqual(this.props, nextProps, (obj, oth, key) => {
+      if (key === 'style') {
+        return shallowEqual(obj, oth)
+      }
+      return undefined
```

Comment:
O wait, I misread this section. This has to be undefined, see docs for shallow equal.

```
If customizer returns undefined (i.e. void 0), then comparisons are handled by the shallowequal function instead
```